framework/src/main/java/org/checkerframework/common/returnsreceiver/FluentAPIGenerator.java

```diff
@@ -0,0 +1,128 @@
+package org.checkerframework.common.returnsreceiver;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.BugInCF;
+import org.checkerframework.javacutil.TypesUtils;
+
+/** Wrapper class for {@link FluentAPIGenerators} Enum to keep it private. */
```

Comment:
Documentation of the public purpose of this class would still be useful... is this meant as an opaque token?

Reply:
The point of the class seems to be method `checkForFluentAPIGenerators`. So I would have liked to see a documentation of that, not the implementation detail about the enum.

Reply:
(I'm also not sure whether I would repeat the class name in the method name... instead of `FluentAPIGenerator.checkForFluentAPIGenerators` isn't `FluentAPIGenerator.check` simpler and as clear?)

Reply:
You could also move the method to the beginning of the class...